db/db_impl.cc

```diff
@@ -1806,7 +1807,9 @@ Status DBImpl::Write(const WriteOptions& options, WriteBatch* my_batch) {
   }
 
   // May temporarily unlock and wait.
-  status = MakeRoomForWrite(my_batch == NULL);
+  if (0==add_and_fetch(&non_block_tickets_, (uint32_t)0))
+      status = MakeRoomForWrite(my_batch == NULL);
```

Comment:
What happens if non_block_tickets_ is non-0, so that we don't call MakeRoomForWrite(), but there's not enough space for all items in the write batch?

That is, is it strictly necessary to call MakeRoomForWrite() before writing the next record, or will the memory footprint just grow beyond the desired limit, at which point blocking writes will be required?